src/files-loader.ts

```diff
@@ -1,35 +1,61 @@
 import * as fs from "fs"
 import * as path from "path"
 import {promisify} from "util"
-import {load as loadMigrationFile} from "./migration-file"
+import {loadMigrationFile} from "./migration-file"
 import {Logger, Migration} from "./types"
 
 const readDir = promisify(fs.readdir)
 
 const isValidFile = (fileName: string) => /\.(sql|js)$/gi.test(fileName)
 
-export const load = async (
+/**
+ * Load the migration files and assert they are reasonably valid.
+ *
+ * 'Reasonably valid' in this case means obeying the file name and
+ * consecutive ordering rules.
+ *
+ * No assertions are made about the validity of the SQL.
+ */
+export const loadMigrationFiles = async (
   directory: string,
-  log: Logger,
+  // tslint:disable-next-line no-empty
+  log: Logger = () => {},
 ): Promise<Array<Migration>> => {
   log(`Loading migrations from: ${directory}`)
 
   const fileNames = await readDir(directory)
   log(`Found migration files: ${fileNames}`)
 
-  if (fileNames != null) {
-    const migrationFiles = [
-      path.join(__dirname, "migrations/0_create-migrations-table.sql"),
-      ...fileNames.map(fileName => path.resolve(directory, fileName)),
-    ].filter(isValidFile)
+  if (fileNames == null) {
+    return []
+  }
 
-    const unorderedMigrations = await Promise.all(
-      migrationFiles.map(loadMigrationFile),
-    )
+  const migrationFiles = [
+    path.join(__dirname, "migrations/0_create-migrations-table.sql"),
+    ...fileNames.map((fileName) => path.resolve(directory, fileName)),
+  ].filter(isValidFile)
 
-    // Arrange in ID order
-    return unorderedMigrations.sort((a, b) => a.id - b.id)
-  }
+  const unorderedMigrations = await Promise.all(
+    migrationFiles.map(loadMigrationFile),
+  )
+
+  // Arrange in ID order
+  const orderedMigrations = unorderedMigrations.sort((a, b) => a.id - b.id)
+
+  validateIdsAreUnique(orderedMigrations)
+
+  return orderedMigrations
+}
 
-  return []
+function validateIdsAreUnique(migrations: Array<Migration>) {
```

Comment:
I added this functionality. Previously ids were validated as unique in the same check that ensured consecutive order. Since we've changed the consecutive order requirement, I wrote this function to ensure that we don't have duplicate ids.